### DISCUSSION TITLE
Easy way to Rename function in PE, while using Debugger?

### DISCUSSION THREAD

**Discussion by Yuvi-App** [Vector35/binaryninja-api#5442] [Q&A] [answered]
Maybe a silly question, but i cannot for the life of me, find a good way of doing this.

In the debugger, I identify functions, and name them as such. However, none of those names transfer over to the PE when the debugger is not running.

Is there a simple way to correlate the current break position in the debugger to the PE? So i can start renaming all the functions in the PE as well? 

Thanks!

**Comment by fuzyll** [comment, marked as answer]
The current implementation of the debugger is that it creates a new Binary View while you are debugging. Changes to this view aren't synchronized with the other Binary Views or saved in any way that lets you keep modifications you make while you are debugging.

This limitation is partly due to Binary Ninja (currently) not having a good mechanism for handling multiple, overlapping regions of memory. We can't represent both the data from the current file _and_ data from the running debugger in the same Binary View, together. We are in the process of addressing this limitation and should hopefully be able to leverage our solution before the end of this year.

In the meantime, you should 👍🏼 and subscribe to [this issue](https://github.com/Vector35/debugger/issues/213), which tracks the problem you're running into.

**Comment by xusheng6** [comment]
I am already working on https://github.com/Vector35/debugger/issues/213, and it is expected to be done in this milestone. I.e., the feature should (hopefully) be available when we release stable 4.1
